proposed/http-middleware/middleware-meta.md

```diff
@@ -222,8 +222,8 @@ and a delegate and must return a response. The middleware may:
 - Evolve the request before passing it to the delegate to execute the next
   available middleware.
 - Evolve the response received from the delegate before returning it.
-- Create and return a response without passing it to the delegate, thereby
-  preventing any further middleware from processing.
+- Create and return a response without further process delegation, thereby
+  preventing any inner middleware from processing.
```

Comment:
> I believe the second middleware cannot be seen as an inner middleware.

Yes, it can. In Stratigility, we represent the pipeline as a queue internally; each successive call to `pipe()` appends to the queue. It acts like an onion: delegating moves to the next layer deeper in the queue. In your example, you end up with the following dispatch flow:

```text
application
  - Hello-World
    - FOO
```

As @Isinlor notes, FOO is inner to Hello-World due to how the developer _composes the pipeline_, which in turn determines execution.

That said, how Stratigility uses delegates is an implementation detail. Not every delegate will be operating in the same way. In fact, we have delegates in Expressive now that simply return responses, and which can be used as "defaults" for when a pipeline completes execution without returning a response. How delegates process a request is entirely up to the library and/or individual developer. Onion-style is common, but not the only possibility.